src/server/journal/journal_slice.cc

```diff
@@ -54,6 +54,8 @@ JournalSlice::JournalSlice() {
 }
 
 JournalSlice::~JournalSlice() {
+  // this mutex is needed to prevent destroying object during change_cb_arr_ executions
```

Comment:
when is the desctructor called? is it called when journal->close is called? this is a thread local

Reply:
yes it's correct